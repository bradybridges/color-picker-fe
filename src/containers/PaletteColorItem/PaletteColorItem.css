.color-container {
  margin: 15px;
  text-align: center;
  width: 8%;
  height: 90%; 
}

.color-code {
  font-size: 1.5em;
}

.color-lock-img {
  width: 25%;
  /* position: relative; */
}