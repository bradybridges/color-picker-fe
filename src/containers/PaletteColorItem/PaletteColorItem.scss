.color-container {
  border-radius: 50%;
  border: solid #fff 1px;
  margin: 15px;
  text-align: center;
  min-width: 200px;
  max-width: 200px;
  height: 200px;
  &:hover {
    cursor: pointer;
  }
}

.color-code {
  font-size: 1.5em;
  margin: 65px 20px 0px 20px;
}

img {
  width: 25%;
  margin: 10px auto;
  /* position: relative; */
}