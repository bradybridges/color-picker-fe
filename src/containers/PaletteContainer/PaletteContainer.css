.palette-container {
  display: flex;
  height: 300px;
  /* padding: 30px; */
  justify-content: center;
}