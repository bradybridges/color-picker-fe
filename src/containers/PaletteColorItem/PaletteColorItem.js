import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setTempPalette } from '../../actions/index';
import './PaletteColorItem.css'

const lockedImg = require('../../images/lock.svg');
const unlockedImg = require('../../images/unlocked-padlock.svg');

export class PaletteColorItem extends Component {
  handleToggleColorLock = () => {
    const { color, isLocked } = this.props;
    const updatedPalette = this.props.tempPalette.map((color) => color);
    const colorToUpdate = updatedPalette.find((palItem) => {
      if(palItem.color === color && palItem.isLocked === isLocked) {
        return true;
      } else { return false; }
    });
    colorToUpdate.isLocked = !colorToUpdate.isLocked;
    this.props.setTempPalette(updatedPalette);
  };

  render() {
    const { color, isLocked } = this.props;
    const lockImg = isLocked ? lockedImg : unlockedImg;
    const style = { backgroundColor: color }
    return (
      <section style={style} className='color-container'>
        <div className='color-container-contents'>
          <p className='color-code'>{color}</p>
          <img src={lockImg} className='color-lock-img' alt="lock or unlock color" onClick={this.handleToggleColorLock}/>
        </div>
      </section>
    );
  }
}

export const mapState = (state) => ({
  tempPalette: state.tempPalette,
});

export const mapDispatch = (dispatch) => ({
  setTempPalette: (palette) => dispatch(setTempPalette(palette)),
});

export default connect(mapState, mapDispatch)(PaletteColorItem);
