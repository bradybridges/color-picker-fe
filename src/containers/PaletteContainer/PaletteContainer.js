import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PaletteColorItem from '../PaletteColorItem/PaletteColorItem';
import './PaletteContainer.css';

export class PaletteContainer extends Component {
  renderColorItems = () => {
    const { tempPalette } = this.props;
    return tempPalette.map((color, i) => {
      return <PaletteColorItem key={`${color}${i}`} color={color.color} isLocked={color.isLocked} />
    });
  }

  render() {
    return (
      <section className="palette-container">
        {this.renderColorItems()}
      </section>
    );
  }
}

export const mapState = (state) => ({
  tempPalette: state.tempPalette,
});

export default connect(mapState)(PaletteContainer);

PaletteContainer.propTypes = {
  tempPalette: PropTypes.array.isRequired,
};
