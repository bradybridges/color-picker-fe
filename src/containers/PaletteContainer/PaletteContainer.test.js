import React from 'react';
import { shallow } from 'enzyme';
import { PaletteContainer, mapState } from './PaletteContainer';
import PaletteColorItem from '../PaletteColorItem/PaletteColorItem';

describe('PaletteContainer', () => {
  const mockTemp = [
    {
      name: 'color_1',
      color: '#FFFFFF',
      isLocked: false,
    },
    {
      name: 'color_2',
      color: '#000000',
      isLocked: false,
    },
    {
      name: 'color_3',
      color: '#CCCCCC',
      isLocked: true,
    },
    {
      name: 'color_4',
      color: '#1F1F1F',
      isLocked: true,
    },
    {
      name: 'color_5',
      color: '#1D1D1D',
      isLocked: true,
    },
  ];
  const wrapper = shallow(<PaletteContainer tempPalette={mockTemp} />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renderColorItems should return an array of PaletteColorItems', () => {
    const result = wrapper.instance().renderColorItems();
    expect(result.length).toEqual(5);
    // expect(result).toEqual();
    // PaletteColorItems are not defined on return?
  });
});

describe('mapState', () => {
  it('should map tempPalette to props', () => {
    const mockTemp = [
      {
        name: 'color_1',
        color: '#FFFFFF',
        isLocked: false,
      },
      {
        name: 'color_2',
        color: '#000000',
        isLocked: false,
      },
      {
        name: 'color_3',
        color: '#CCCCCC',
        isLocked: true,
      },
      {
        name: 'color_4',
        color: '#1F1F1F',
        isLocked: true,
      },
      {
        name: 'color_5',
        color: '#1D1D1D',
        isLocked: true,
      },
    ];
    const expectedState = { tempPalette: mockTemp };
    const mockState = {
      error: 'Something terrible happened',
      tempPalette: mockTemp,
    };
    const mappedState = mapState(mockState);
    expect(mappedState).toEqual(expectedState);
  });
});
