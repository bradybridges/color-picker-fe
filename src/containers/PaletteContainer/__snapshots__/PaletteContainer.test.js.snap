// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteContainer should match snapshot 1`] = `
<section
  className="palette-container"
>
  <Connect(PaletteColorItem)
    color="#FFFFFF"
    isLocked={false}
    key="[object Object]0"
  />
  <Connect(PaletteColorItem)
    color="#000000"
    isLocked={false}
    key="[object Object]1"
  />
  <Connect(PaletteColorItem)
    color="#CCCCCC"
    isLocked={true}
    key="[object Object]2"
  />
  <Connect(PaletteColorItem)
    color="#1F1F1F"
    isLocked={true}
    key="[object Object]3"
  />
  <Connect(PaletteColorItem)
    color="#1D1D1D"
    isLocked={true}
    key="[object Object]4"
  />
</section>
`;
